packages/server/src/api/controllers/query/index.ts

```diff
@@ -100,10 +99,18 @@ export async function save(ctx: UserCtx<Query, Query>) {
   const datasource = await sdk.datasources.get(query.datasourceId)
 
   let eventFn
-  if (!query._id) {
+  if (!query._id && !query._rev) {
```

Comment:
Will we ever have rev but not I'd?

Reply:
if we have an ID it is likely we will have a rev, but just for full confirmation making sure we have both.